### PR TITLE
Add ODR mock controller AB#10908

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
# Fixes or Implements [AB#10908](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10908)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Adds ODR mock controller.
Fixes protective word modal not being displayed.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES | NO

### Browsers Tested

-   [ ] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

File: Apps/Encounter/src/Delegates/RestMSPVisitDelegate.cs [hunk 1]
```diff
@@ -108,7 +109,7 @@ public async Task<RequestResult<MSPVisitHistoryResponse>> GetMSPVisitHistoryAsyn
                 try
                 {
                     string json = JsonSerializer.Serialize(request, options);
-                    using HttpContent content = new StringContent(json);
+                    using HttpContent content = new StringContent(json, null, MediaTypeNames.Application.Json);
```
Comment (tiagograf-AOT) [inline Apps/Encounter/src/Delegates/RestMSPVisitDelegate.cs:112]: content type was being set as plain/text, updated to the correct "app/json", ODR works with both

File: Apps/WebClient/src/ClientApp/src/store/modules/medication/modules/statement/actions.ts [hunk 2]
```diff
@@ -54,10 +54,12 @@ export const actions: MedicationStatementActions = {
                             );
                             reject(result.resultError);
                         } else {
-                            EventTracker.loadData(
-                                EntryType.Medication,
-                                result.resourcePayload.length
-                            );
+                            if (result.resultStatus === ResultType.Success) {
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/store/modules/medication/modules/statement/actions.ts:57]: it was erroring out when "ActionRequired" protective word was returned

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/store/modules/medication/modules/statement/actions.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/store/modules/medication/modules/statement/actions.ts:57]: good catch!

File: Apps/Mock/src/Controllers/OdrController.cs [hunk 3]
```diff
@@ -25,14 +32,80 @@ namespace HealthGateway.Mock.Controllers
...
+            else
+            {
+                variables.Add("${value}", string.Empty);
+            }
+            string? payload = AssetReader.Read("HealthGateway.Mock.Assets.ProtectiveWord.json");
+            return new ContentResult { Content = ReplaceVariables(payload!, variables), ContentType = "application/json" };
+        }
+
+
+        private static string ReplaceVariables(string payload, Dictionary<string, string> variables)
```
Comment (FuriousLlama) [inline Apps/Mock/src/Controllers/OdrController.cs:101]: we might want to make this accessible to other controllers

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**